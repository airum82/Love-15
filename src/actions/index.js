export const fetchCourts = (nearbyCourts) => ({
  type: 'COURTS',
  nearbyCourts
})

export const createAccount = (accountInfo) => ({
  type: 'CREATE_ACCOUNT',
  accountInfo
})

export const logIn = (accountInfo) => ({
  type: 'LOG_IN',
  accountInfo
})