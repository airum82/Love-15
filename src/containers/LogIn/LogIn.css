.log-in form {
  margin: 0 auto;
  font-size: 1.2em;
}

.log-in {
  text-align: center;
}

.log-in input {
  height: 4vh;
  margin: 1% auto;
  display: block;
  font-size: 1.3em;
}

.log-in button {
  margin: 1% auto;
  padding: 1% 1%;
  border: none;
  background-color: #60b0f4;
  border-radius: 25px;
  width: 21%;
  color: white;
  font-size: 1.2em;
}

.log-in button:hover {
  cursor: pointer;
}

.create-link {
  font-size: 1.3em;
  color: black;
  font-weight: 700;
}

.create-link:hover {
  color: white;
}

.log-in-button {
  display: block;
  text-align: right;
  margin-right: 3%;
}