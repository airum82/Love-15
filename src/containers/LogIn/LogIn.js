import React, { Component } from 'react';
import { connect } from 'react-redux';
import { logIn } from '../../actions';
import PropTypes from 'prop-types';

export class LogIn extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: ''
    }
  }

  handleAccountEntry = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  render() {
    return (
      <div className="log-in">
        <form 
          onChange={this.handleAccountEntry}
          onSubmit={(e) => {
            e.preventDefault();
            this.props.handleLogIn(this.state);
          }}
        >
          <input type="email" name="email" />
          <input type="password" name="password" />
          <button>Log In</button>
        </form>
      </div>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  handleLogIn: (accountInfo) => dispatch(logIn(accountInfo))
})

export default connect(null, mapDispatchToProps)(LogIn);