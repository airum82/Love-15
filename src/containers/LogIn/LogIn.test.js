import { LogIn, mapDispatchToProps } from './LogIn';
import { shallow, mount} from 'enzyme';
import React, { Component } from 'react';
import { logIn } from '../../actions';

describe('LogIn', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<LogIn />);
  })

  it('should match snapshot upon render', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should call handleAccountEntry on change to form', () => {
    wrapper = mount(<LogIn />);
    const spy = jest.spyOn(wrapper.instance(), 'handleAccountEntry');
    wrapper.find('form').simulate('change');
    wrapper.find('form').simulate('change');
    expect(spy).toHaveBeenCalled();
  })

  it('handleAccountEntry should update state with user input', () => {
    const initialState = {
      email: '',
      password: ''
    }
    const mockEvent = {
      target: {
        name: 'email',
        value: 'jim@jim.com'
      }
    }
    const updatedState = {
      email: 'jim@jim.com',
      password: ''
    }
    expect(wrapper.state()).toEqual(initialState);
    wrapper.instance().handleAccountEntry(mockEvent);
    expect(wrapper.state()).toEqual(updatedState);
  })

  it('should call handleLogIn on form submit with correct params', () => {
    const mockHandleLogIn = jest.fn()
    wrapper = mount(<LogIn handleLogIn={mockHandleLogIn} />);
    const state = wrapper.state();
    wrapper.find('form').simulate('submit');
    expect(mockHandleLogIn).toHaveBeenCalledWith(state);
  })
})

describe('mapDispatchToProps', () => {

  it('handleLogIn should call dispatch with correct params', () => {
    const mockDispatch = jest.fn();
    const mappedProps = mapDispatchToProps(mockDispatch);
    const mockAccountInfo = {
      email: 'jamie@this.com',
      password: 'eimaj'
    }
    mappedProps.handleLogIn(mockAccountInfo);
    expect(mockDispatch).toHaveBeenCalledWith(logIn(mockAccountInfo));
  })
})