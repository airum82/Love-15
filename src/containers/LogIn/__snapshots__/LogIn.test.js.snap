// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogIn should match snapshot upon render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LogIn />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onChange={[Function]}
        onSubmit={[Function]}
>
        <input
                name="email"
                type="email"
        />
        <input
                name="password"
                type="password"
        />
        <button>
                Log In
        </button>
</form>,
      "className": "log-in",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="email"
            type="email"
/>,
          <input
            name="password"
            type="password"
/>,
          <button>
            Log In
</button>,
        ],
        "onChange": [Function],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "email",
            "type": "email",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "password",
            "type": "password",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Log In",
          },
          "ref": null,
          "rendered": "Log In",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onChange={[Function]}
          onSubmit={[Function]}
>
          <input
                    name="email"
                    type="email"
          />
          <input
                    name="password"
                    type="password"
          />
          <button>
                    Log In
          </button>
</form>,
        "className": "log-in",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="email"
              type="email"
/>,
            <input
              name="password"
              type="password"
/>,
            <button>
              Log In
</button>,
          ],
          "onChange": [Function],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "email",
              "type": "email",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "password",
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Log In",
            },
            "ref": null,
            "rendered": "Log In",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
